docs/cookbook/create-basic-grid.md

```diff
@@ -0,0 +1,270 @@
+
+We need to inject (pass through constructor) `ArticleGridFactory` created earlier and pass grid view to the template.
+
+```php
+class DashboardController extends AdminBaseController
+{
+    ...
+
+    /**
+     * @Route("/dashboard/articles/")
+     */
+    public function articlesAction()
+    {
+        $grid = $this->awesomeGridFactory->create();
+
+        return $this->render('@ShopsysShop/Admin/Content/Dashboard/articles.html.twig', [
+            'gridView' => $grid->createView(),
+        ]);
+    }
+```
+
+In the template, we render the whole grid just easy by calling `{{ gridView.render() }}` and grid will start do its work.
+
+![Display Data with Grid](img/grid-display-data.png)
+
+## Modify the basic grid appearance
+As you probably noticed, dates in the third column are not printed much friendly.
+To adjust appearance, we just need to extend the default grid template and modify it accordingly.
+
+We create the new twig template `articleGrid.html.twig` in `src/Shopsys/ShopBundle/Resources/views/Admin/Content`
+The template have to extend `@ShopsysFramework/Admin/Grid/Grid.html.twig` and override block `grid_value_cell_id_createdAt` <!--- TODO add after page will exist ([you can read more about blocks here](/docs/administration/grid-rendering.md)) -->
+
+```twig
+{% extends '@ShopsysFramework/Admin/Grid/Grid.html.twig' %}
+
+{% block grid_value_cell_id_createdAt %}
+    {{ value|formatDate }} {# pass value from the cell through helper #}
+{% endblock %}
+```
+
+Now that we have template ready, we just need to set it as the theme in grid factory.
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
+        ...
+
++       $grid->setTheme('@ShopsysShop/Admin/Content/Dashboard/articleGrid.html.twig');
+
+        return $grid;
+    }
+```
+
+## Sort rows
+Now we see data as we want, but it would be nice to be able to adjust the view from the users perspective.
+We may want to ease finding of a certain article by allowing to sort rows by name or date of creation.
+
+A default order will be by date with newest articles at the top.
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
+        $grid->addColumn('id', 'a.id', t('Id'));
+-       $grid->addColumn('name', 'a.name', t('Title'));
++       $grid->addColumn('name', 'a.name', t('Title'), true);
+-       $grid->addColumn('createdAt', 'a.createdAt', t('Created'));
++       $grid->addColumn('createdAt', 'a.createdAt', t('Created'), true);
+
++       $grid->setDefaultOrder('createdAt', DataSourceInterface::ORDER_DESC);
+    ...
+```
+
+## Paginate results
+As the number of articles grow, the clarity decreases rapidly.
+To keep work with grid enjoyable, we can split data across several pages with pagination.
+
+In the grid is just a matter of calling one method.
```

Comment:
Hm, a simple question might come to the mind of a reader here, how many rows are displayed on a single page? And am I able to change this? :innocent: